src/vs/editor/contrib/suggest/media/suggest.css

```diff
@@ -229,16 +229,33 @@
 	padding: 4px 0 4px 5px;
 }
 
+.monaco-editor .suggest-widget .details > .monaco-scrollable-element > .body > .header > .type {
+	padding-bottom: 12px;
+}
+
 .monaco-editor .suggest-widget .details > .monaco-scrollable-element > .body > .docs {
 	margin: 0;
 	padding: 4px 5px;
 	white-space: pre-wrap;
 }
 
 .monaco-editor .suggest-widget .details > .monaco-scrollable-element > .body > .docs.markdown-docs {
+	padding: 0;
 	white-space: initial;
 }
 
+.monaco-editor .suggest-widget .details > .monaco-scrollable-element > .body > .docs.markdown-docs > div {
+	padding: 4px 5px;
```

Comment:
Last question :)

Why are we moving the padding from `.docs.markdown-docs` to `.docs.markdown-docs > div`?

Reply:
Hehe, no problem!

I'll try to explain it:

Problem:

1. We need this `padding-bottom` in order to provide space between the widget's header and content:

<img width="794" alt="screen shot 2018-10-11 at 11 46 03 pm" src="https://user-images.githubusercontent.com/13837769/46845006-e208d280-cdaf-11e8-9b9e-d2c9c6b4719f.png">

2. We also need the `.markdown-docs` `padding` to provide space between the content and the widget borders:

<img width="791" alt="screen shot 2018-10-11 at 11 46 13 pm" src="https://user-images.githubusercontent.com/13837769/46845007-e208d280-cdaf-11e8-965b-b1bcb7b4b333.png">

3. When there is no content to show, the `.markdown-docs` `padding` generates this unwanted effect:

<img src="https://user-images.githubusercontent.com/13837769/46646012-243ad580-cb5e-11e8-9a45-015beccdc416.png" />

Solution:

When the suggestion docs has `markdown` content, this function is being called in order to render it's contents:

https://github.com/Microsoft/vscode/blob/e8edae54aa99d06ddb37211bebc6faffb20ff1fb/src/vs/editor/contrib/suggest/suggestWidget.ts#L262

This one ends up calling the `createElement` in the `htmlContentRenderer` class, which eventually generates the html element that will be added as the `.markdown-docs` children:

https://github.com/Microsoft/vscode/blob/e8edae54aa99d06ddb37211bebc6faffb20ff1fb/src/vs/base/browser/htmlContentRenderer.ts#L30

When there is no content to show, the `options.inline` property becomes `true` and an empty `span` is added to the `.markdown-docs`. Here we shouldn't add `padding`. 

In order to consider this scenario and avoid adding the problematic `padding` we can apply the `padding` to the `.markdown-docs`'s  immediate children instead of applying it to the `.markdown-docs` element itself so we can discriminate between this two cases:

1. the children is a `div` element and we should always add padding
2. the children is a `span` element and we should only add padding it the `span` is not empty  (I didn't found any suggestion widget with `markdown-docs` and a non-empty `span` as children but I think we should handle it, just in case).

That's why I moved the padding from `.docs.markdown-docs` to `.docs.markdown-docs > div` (and also added another rule to handle the span:non(:empty) case).

Hope I was able to explain my self in a clear way.

Kr